src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -84,4 +104,29 @@ private void doPostCleanup(BibEntry entry) {
             return Collections.emptyList();
         }
     }
+
+    /**
+     * Returns registration agency. Optional.empty() if no agency is found.
+     *
+     * @param doi the doi to be searched
+     */
+    public Optional<String> getAgency(DOI doi) throws IOException {
```

Comment:
Please move this method to the `DOI` class (i..e the call should look like `doi.getAgency()`)

Reply:
I was not sure where to put it in the beginning, but it makes sense to move to DOI class.